### PR TITLE
Add Vision Transformer demo for image classification (Fixes #13372)

### LINKED ISSUE #13372: Add Vision Transformer (ViT) Demo to computer_vision Module
### Feature description

This PR adds a fully functional demo of a Vision Transformer (ViT) for image classification using Hugging Face Transformers.

Features included:
- Loads a sample image from the web.
- Uses ViTImageProcessor for preprocessing.
- Performs inference with ViTForImageClassification.
- Prints the predicted label for the image.
- Handles network timeout and correct import order.

Example output:
Predicted label: tabby, tabby cat

This demo can be used as a reference for anyone learning ViT or image classification with Hugging Face Transformers.

Additional Notes:
- Requires torch, transformers, and PIL.
- Can be extended to classify local images easily.

Hacktoberfest 2025: ✅

### PR DESCRIPTION
**Pull request by kdt523**
This pull request adds a Vision Transformer (ViT) demo for image classification to the computer_vision module. The new script demonstrates how to use a pre-trained ViT model from Hugging Face Transformers to classify images from either a URL or a local file. The implementation includes full type hints, comprehensive docstrings, and working doctests that pass automated testing. The code follows all repository conventions for naming, structure, and documentation, and includes references to the original ViT paper and Hugging Face documentation.
References:
[https://arxiv.org/abs/2010.11929](url)
[https://huggingface.co/docs/transformers/model_doc/vit](url)
Fixes: #13372

### CODE REVIEW

File: computer_vision/vision_transformer_demo.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+    result = {
+        "label": predictions[0]["label"],
+        "score": predictions[0]["score"],
+        "top_k_predictions": predictions if top_k > 1 else None,
+    }
+
+    return result
+
+
+def main() -> None:
```
[the remark this answers is not in the document]
Comment (kdt523) [reply computer_vision/vision_transformer_demo.py:174]: @algorithms-keeper 

Thank you for the review! I've added a doctest to the `main()` function. The test verifies the function is callable. I used `+SKIP` for the execution test since `main()` requires network access and model downloads.



Let me know if you need any other changes.

**Comment by kdt523** [comment]
Hi @TheAlgorithms/maintainers 
This PRAdd Vision Transformer demo for image classification (Fixes #13372) #13538 all checks have passed successfully.  
Could someone with write access please review and approve it when you get a chance?  
Thanks a lot for your time and help!

File: computer_vision/vision_transformer_demo.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+        ...     load_image("nonexistent_file.jpg")
+        ... except FileNotFoundError:
+        ...     print("File not found")
+        File not found
+    """
+    if isinstance(image_source, (str, Path)) and str(image_source).startswith(
+        ("http://", "https://")
+    ):
+        try:
+            with httpx.Client(timeout=timeout) as client:
```
Comment (cclauss) [inline computer_vision/vision_transformer_demo.py]:
```suggestion
            with httpx2.Client(timeout=timeout) as client:
```

File: computer_vision/vision_transformer_demo.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+                response = client.get(str(image_source))
+                response.raise_for_status()
+                return Image.open(BytesIO(response.content)).convert("RGB")
+        except httpx.TimeoutException:
```
Comment (cclauss) [inline computer_vision/vision_transformer_demo.py]:
```suggestion
        except httpx2.TimeoutException:
```

File: computer_vision/vision_transformer_demo.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+            msg = (
+                f"Request timed out after {timeout} seconds. "
+                "Try increasing the timeout parameter."
+            )
+            raise TimeoutError(msg)
+        except httpx.HTTPError as e:
```
Comment (cclauss) [inline computer_vision/vision_transformer_demo.py]:
```suggestion
        except httpx2.HTTPError as e:
```

File: computer_vision/vision_transformer_demo.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+Vision Transformers apply the transformer architecture (originally designed for NLP)
+to computer vision by splitting images into patches and processing them with
+self-attention mechanisms.
+
+Requirements:
+    - torch
+    - transformers
+    - Pillow (PIL)
+    - httpx (already in repo dependencies)
```
Comment (cclauss) [inline computer_vision/vision_transformer_demo.py]:
```suggestion
    - httpx2 (already in repo dependencies)
```

File: computer_vision/vision_transformer_demo.py [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+
+from __future__ import annotations
+
+import sys
+from io import BytesIO
+from pathlib import Path
+from typing import Any
+
+try:
+    import httpx
```
Comment (cclauss) [inline computer_vision/vision_transformer_demo.py]:
```suggestion
    import httpx2
```

**Comment by cclauss** [comment]
No PEP 723 header for dependencies.
This is a how-to-use, not an algorithm.  See CONTRIBUTING.md.
